src/networking.c

```diff
@@ -117,6 +117,57 @@ int authRequired(client *c) {
     return auth_required;
 }
 
+sds getAllClientMetaFields(dict *meta, sds scaping) {
+    dictIterator *di = dictGetIterator(meta);
+    dictEntry *de;
+    sds result = sdsempty();
+
+    while((de = dictNext(di)) != NULL) {
+        sds key = dictGetKey(de);
+        sds value = dictGetVal(de);
+
+        result = sdscatfmt(result,
+            "%s=%s%s",
+            key,
+            value,
+            scaping
+        );
+    }
+
+    return result;
+}
+
+sds getClientMetaFields(client *c) {
+    int i;
+    sds result = sdsempty();
+
+    for (i = 2; i < c->argc; i++) {
+        dictEntry *de;
+        de = dictFind(c->meta, c->argv[i]->ptr);
+        if (de == NULL) {
+            result = sdscatfmt(result,
+                "%s= ",
+                c->argv[i]->ptr
+            );
+            continue;
+        };
+        sds value = dictGetVal(de);
+        if (value == NULL) {
```

Comment:
I wouldn't expect value to ever be NULL. Would suggest either asserting it's not null or omitting this check.